lib/Core/Memory.h

```diff
@@ -399,6 +399,9 @@ class PersistentState : public ObjectState {
     /// Create a new persistent object state from the given non-persistent
     /// object state and symbolic bool array of cache lines. Also requires
     /// a symbolic void* array (int64) for root cause.
+    ///
+    /// We now take the state to add the array names to state.arrayNames and 
+    /// fail if any arrays we create internally already exist.
```

Comment:
Could also just have a static set of names in PersistentState

Reply:
Yeah, but then you could technically run into a nasty bug if something other module tried to use the same name...this isn't too inconvenient.